### PR TITLE
Introduction to sessions

### PR DESCRIPTION
**Pull request by konqi**
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1241)

### CODE REVIEW

File: ratpack-manual/src/content/chapters/13-http.md [hunk 1]
```diff
@@ -513,4 +513,128 @@ Support for rendering different representations of a resource (JSON/XML/HTML, GI
...
+		module(SessionModule)
+	}
+```
+
+### Use session
+
+You are now set for sessions.
+The following is a simple example of an application that uses sessions.
+
+```language-java
```
Comment (beckje01) [inline ratpack-manual/src/content/chapters/13-http.md]: Can you add `tested` after `language-java` it should then test that the code is correct in the manual.

File: ratpack-manual/src/content/chapters/13-http.md [hunk 2]
```diff
@@ -513,4 +513,128 @@ Support for rendering different representations of a resource (JSON/XML/HTML, GI
...
+		get('stop') { Session session ->
+			session.terminate().then {
+				response.send('The session is dead, dead, dead.')
+			}
+		}
+	}
+}
+```
+
+All session operations return either a `Promise` or an `Operation`.
```
Comment (beckje01) [inline ratpack-manual/src/content/chapters/13-http.md]: Make links to the api doc?

File: ratpack-manual/src/content/chapters/13-http.md [hunk 3]
```diff
@@ -513,4 +513,128 @@ Support for rendering different representations of a resource (JSON/XML/HTML, GI
...
+Whenever you first add data to the session the id is generated and the cookie will be sent to the client using the `Add-Cookie` header in the response.
+Therefor you need to make sure that you a response is sent to the client or the session will be lost.
+Consecutive client requests contain the cookie in the `Cookie`-Header, so ratpack can determine the client's session.
+
+Here are some useful links to dive deeper into ratpack session handling:
+- more samples of how sessions work in the [ratpack-session module tests](https://github.com/ratpack/ratpack/blob/master/ratpack-session/src/test/groovy/ratpack/session/SessionSpec.groovy)
+- [ratpack javadoc](https://ratpack.io/manual/current/api/ratpack/session/SessionModule.html) contains lots of examples and information
+- cookie behaviour can be customized (i.e. change name of cookie, use custom ids/expiration date) by providing a custom [SessionCookieConfig](https://ratpack.io/manual/current/api/ratpack/session/SessionCookieConfig.html) to the session module 
+
+### A final note when you use the default session store (in-memory):
```
Comment (beckje01) [inline ratpack-manual/src/content/chapters/13-http.md]: This section should be in the voice of the manual not personal findings.

File: ratpack-manual/src/content/chapters/13-http.md [hunk 4]
```diff
@@ -513,4 +513,128 @@ Support for rendering different representations of a resource (JSON/XML/HTML, GI
...
+
+Make sure you terminate an old session before starting a new one (see `get('start')`-handler).
+This way you make sure that you actually get a new session and don't just add data to an existing session.
+
+### Limitations 
+
+The ratpack session module uses in-memory sessions by default.
+It can hold up to 1000 sessions at a time and will drop the oldest session if a new session is opened.
+If you expect more than 1000 sessions you should consider to use a different session store than the default module.
+You could for example use the `ratpack-session-redis` module if you have a redis server handy.
```
Comment (beckje01) [inline ratpack-manual/src/content/chapters/13-http.md]: Call out the client side session module as well.

**Comment by beckje01** [comment]
Overall great work! Please also add yourself to the Contributors section https://github.com/ratpack/ratpack/blob/master/ratpack-manual/src/content/chapters/99-about-the-project.md#contributors

File: ratpack-manual/src/content/chapters/13-http.md (hunk 4 above)
Comment (konqi) [inline ratpack-manual/src/content/chapters/13-http.md]: 👍

File: ratpack-manual/src/content/chapters/13-http.md (hunk 3 above)
Comment (konqi) [inline ratpack-manual/src/content/chapters/13-http.md]: 👍

**Comment by konqi** [comment]
Thanks @beckje01. The snippets should now pass testing.

**Comment by beckje01** [comment]
Waiting on CI

**Pull request merged**